pkg/server/routes.go

```diff
@@ -81,6 +82,7 @@ func (srv *RegistrationServer) SetupRoutes() error {
 		securedV1 := srv.router.Group("/api/v1")
 		securedV1.Use(authMiddleware.HandlerFunc())
 		securedV1.POST("/signup", signupCtrl.PostHandler)
+		securedV1.GET("/signupcheck", signupCheckCtrl.GetHandler)
```

Comment:
```suggestion
		securedV1.GET("/signupcheck", signupCtrl.StatusGetHandler)
```

Let's add another handler to the existing signup controller instead of introducing a new one: `/api/v1/signup/status`.
The payload it returns also could be a bit more detailed:

```
type SignupStatus struct {
    Ready bool
    Reason string
    Message *string
}
```

`Ready == true` only when the account is ready to be used by the user.
`Reason` can be: `pendingApproval`, `provisioning`, 'provisioned'
`Message` is optional. It's a human readable message if there is some problem with the account/signup.

Basically it's a simplified version of Signup/MUR status/conditions sub resource we use in Kube.

Reply:
We could also use the same endpoint `/api/v1/signup` but with `GET`. I'd say this would even better match the pattern. WDYT?

Reply:
Added the above extra data in 7d2cd53.

Reply:
+1 it should be `GET /api/v1/signup `

Reply:
@michaelkleinhenz it was the initial idea to use `GET /api/v1/signup` to get the signup with status. But we could move status to a subresource `GET /api/v1/signup/status` if we want to minimize the payload. I'm fine with both options.